test/helper/webauth/index.ts

```diff
@@ -48,6 +48,7 @@ export default async function authorize(
   password: string,
 ) {
   const browser = await puppeteer.launch({
+    headless: true,
```

Comment:
```
Puppeteer old Headless deprecation warning:
        In the near future `headless: true` will default to the new Headless mode
        for Chrome instead of the old Headless implementation. For more
        information, please see [https://developer.chrome.com/articles/new-headless/.](https://developer.chrome.com/articles/new-headless/)
        Consider opting in early by passing `headless: "new"` to `puppeteer.launch()`
        If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.

      48 |   password: string,
      49 | ) {
    > 50 |   const browser = await puppeteer.launch({
         |                                   ^
      51 |     args: ['--no-sandbox', '--disable-setuid-sandbox'],
      52 |   });
      53 |   let ret;

      at ChromeLauncher.launch (node_modules/puppeteer-core/src/node/ChromeLauncher.ts:54:15)
      at PuppeteerNode.launch (node_modules/puppeteer-core/src/node/PuppeteerNode.ts:184:27)
      at launch (test/helper/webauth/index.ts:50:35)
      at Object.<anonymous> (test/connection-session.test.ts:164:37)
```

https://github.com/jsforce/jsforce/actions/runs/6909407649/job/18800708365#step:8:266